src/reducers/ReportsReducer.tsx

```diff
@@ -476,6 +480,44 @@ export const reportsReducer = (state: ReportState = initialState, action: Generi
             return nextState;
         }
 
+        // FETCH_REPORT_PAYLOAD_FILE
```

Comment:
From my total lack of knowledge , could this lines on all usecases be refactored and extracted to a method to avoid code duplication ?

Reply:
@jonathanvila The only thing that comes to my mind to enhance this code is to create a method like this:
```
export const updateObject = (oldObject: any, updatedProperties: any) => {
    return {
        ...oldObject,
        ...updatedProperties
    };
};
```

And replace:
```
const nextState: ReportState = {
    ...state,
    reportPayloadFileFetchStatus: {
        ...state.reportPayloadFileFetchStatus,
        error: null,
        status: 'inProgress'
    }
};
```

By:
```
const nextState: ReportState = updateObject(state, {
    reportPayloadFileFetchStatus: {
        ...state.reportPayloadFileFetchStatus,
        error: null,
        status: 'inProgress'
    }
});
```

But it reduced the code in just 1 line, I still need to copy the rest of the fields manually.

While thinking I also realized that if we create generic methods to avoid duplication code we will end up having also generic objects and the IDE won't be able to check the consistency of objects. In the example below, you will see that the current way of creating objects allows the IDE to alert developers about some invalid FIELDs. So the current way of coding might duplicate code but helps the IDE. I would suggest keeping the current way of coding; however, I'm open to suggestions.

![Screenshot from 2020-01-23 12-05-58](https://user-images.githubusercontent.com/2582866/72979737-48650900-3dd9-11ea-83c7-65d04ccfecab.png)